xbmc/music/MusicDatabase.cpp

```diff
@@ -2832,6 +2832,14 @@ void CMusicDatabase::GetFileItemFromDataset(const dbiplus::sql_record* const rec
   ReplayGain replaygain;
   replaygain.Set(record->at(song_strReplayGain).get_asString());
   item->GetMusicInfoTag()->SetReplayGain(replaygain);
+  // Get the number of discs using dataset2 as m_pDS is already in use
+  std::string strSQL = PrepareSQL("SELECT iDiscTotal FROM album WHERE album.idAlbum = %i",
+                                  record->at(song_idAlbum).get_asInt());
+  if(m_pDS2->query(strSQL))
+  {
+    item->GetMusicInfoTag()->SetTotalDiscs(m_pDS2->fv(0).get_asInt());
+    m_pDS2->close();
+  }
```

Comment:
Not a good way to implement this, will mean an extra query per song when showing the songs node.